app/controllers/merchant/discounts_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Merchant::DiscountsController < ApplicationController
+  def index
+    @discounts = current_user.merchants.first.discounts
+  end
+
+  def show
+    @discount = Discount.find(params[:id])
+  end
+
+  def new
+    @discount = Discount.new
+  end
+
+  def create
+    merchant = current_user.merchants.first
+    @discount = merchant.discounts.create(discount_params)
```

Comment:
next step is to refactor this controller, these actions are doing too many things